syntax = "proto3";

package antdsb;

import "google/protobuf/empty.proto";

service BotService {
  rpc GetStatus (StatusRequest) returns (StatusReply);
  rpc SendMessage (SendMessageRequest) returns (SendMessageReply);

  rpc StartBot (google.protobuf.Empty) returns (StatusReply);
  rpc StopBot (google.protobuf.Empty) returns (StatusReply);
  rpc RestartBot (google.protobuf.Empty) returns (StatusReply);
}

message StatusRequest {}

message StatusReply {
  bool running = 1;
  string last_error = 2;
}

message SendMessageRequest {
  uint64 channel_id = 1;
  string text = 2;
}

message SendMessageReply {
  bool ok = 1;
  string error = 2;
}
